File: app/src/main/java/com/disruption/cookcentral/ui/favs/FavouritesViewModel.java
package com.disruption.cookcentral.ui.favs;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.disruption.cookcentral.data.RecipeDatabase;
import com.disruption.cookcentral.data.RecipeRepository;
import com.disruption.cookcentral.models.CachedRecipe;

import java.util.List;

public class FavouritesViewModel extends AndroidViewModel {

    public LiveData<List<CachedRecipe>> mFavData;
    private RecipeRepository mRecipeRepository;

    public FavouritesViewModel(@NonNull Application application) {
        super(application);
        mRecipeRepository = new RecipeRepository(RecipeDatabase.getInstance(application).recipeDao());
        mFavData = mRecipeRepository.getAllFavs();
    }

    public void deleteRecipeFromFavourites(CachedRecipe recipe) {
        mRecipeRepository.deleteRecipeFromFavs(recipe);
    }

    public void nukeFavsTable() {
        mRecipeRepository.nukeFavsTable();
    }

    public void insertRecipeToFavourites(CachedRecipe recipe) {
        mRecipeRepository.addRecipeToFavs(recipe);
    }
}

File: app/src/main/java/com/disruption/cookcentral/ui/favs/FavouritesViewModel.kt
package com.disruption.cookcentral.ui.favs

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.disruption.cookcentral.data.RecipeRepository
import com.disruption.cookcentral.models.CachedRecipe
import org.koin.core.KoinComponent
import org.koin.core.inject

class FavouritesViewModel(application: Application) : AndroidViewModel(application), KoinComponent {

    private val mRecipeRepository: RecipeRepository by inject()

    val mFavData = mRecipeRepository.allFavs

    fun deleteRecipeFromFavourites(recipe: CachedRecipe) {
        mRecipeRepository.deleteRecipeFromFavs(recipe)
    }

    fun nukeFavsTable() {
        mRecipeRepository.nukeFavsTable()
    }

    fun insertRecipeToFavourites(recipe: CachedRecipe) {
        mRecipeRepository.addRecipeToFavs(recipe)
    }
}